lib/jazzy/doc_builder.rb

```diff
@@ -165,12 +170,25 @@ def self.write_undocumented_file(undocumented, output_dir)
             d['key.modulename'] || ''
           end
         end
+        
+        warnings = []
         tokens_by_file.each_key do |file|
-          f.write(file + "\n")
           tokens_by_file[file].each do |token|
-            f.write("\t" + decl_for_token(token) + "\n")
+            warnings << {
+              file: token['key.filepath'],
+              line: token['key.doc.line'],
+              symbol: token['key.name'],
+              symbol_kind: token['key.kind'],
+              warning: 'undocumented'
+            }
           end
         end
+
+        lint_report = {
+          warnings: warnings,
+          source_directory: options.source_directory
```

Comment:
The linting engine I'm using calls a linter once for each file that was changed. I use `source_directory` to avoid re-running jazzy on the same project multiple times.